python/kiss_icp/tools/visualizer.py

```diff
@@ -20,223 +20,301 @@
 from abc import ABC
-from functools import partial
-from typing import Callable, List
 
 import numpy as np
 
-CYAN = np.array([0.24, 0.898, 1])
-RED = np.array([128, 0, 0]) / 255.0
-BLACK = np.array([0, 0, 0]) / 255.0
-BLUE = np.array([0.4, 0.5, 0.9])
-GRAY = np.array([0.4, 0.4, 0.4])
-SPHERE_SIZE = 0.20
+from kiss_icp.config.parser import KISSConfig
+
+BACKGROUND_COLOR = [0.8470, 0.8588, 0.8863]
+FRAME_COLOR = [0.2156, 0.2471, 0.3176]
+KEYPOINTS_COLOR = [0.8588, 0.3294, 0.3803]
+LOCAL_MAP_COLOR = [0.3451, 0.6431, 0.6902]
+TRAJECTORY_COLOR = [0.1058, 0.1059, 0.1176]
+FRAME_PTS_SIZE = 0.2
+KEYPOINTS_PTS_SIZE = 0.3
+MAP_PTS_SIZE = 0.1
 
 
 class StubVisualizer(ABC):
-    def __init__(self):
+    def __init__(self, config: KISSConfig):
         pass
 
-    def update(self, source, keypoints, target_map, pose):
+    def update(self, source, keypoints, target_map, pose, lest_time):
         pass
 
 
-class RegistrationVisualizer(StubVisualizer):
-    # Public Interaface ----------------------------------------------------------------------------
-    def __init__(self):
-        try:
-            self.o3d = importlib.import_module("open3d")
-        except ModuleNotFoundError as err:
-            print(f'open3d is not installed on your system, run "pip install open3d"')
-            exit(1)
+def start_pause_callback():
+    button_name = "PAUSE" if Kissualizer.play_mode else "START"
+    if Kissualizer.polyscope.imgui.Button(button_name) or Kissualizer.polyscope.imgui.IsKeyPressed(
+        Kissualizer.polyscope.imgui.ImGuiKey_Space
+    ):
+        Kissualizer.play_mode = not Kissualizer.play_mode
 
-        # Initialize GUI controls
-        self.block_vis = True
-        self.play_crun = False
-        self.reset_bounding_box = True
 
-        # Create data
-        self.source = self.o3d.geometry.PointCloud()
-        self.keypoints = self.o3d.geometry.PointCloud()
-        self.target = self.o3d.geometry.PointCloud()
-        self.frames = []
+def next_frame_callback():
+    if Kissualizer.polyscope.imgui.Button("NEXT FRAME") or Kissualizer.polyscope.imgui.IsKeyPressed(
+        Kissualizer.polyscope.imgui.ImGuiKey_N
+    ):
+        Kissualizer.block_execution = not Kissualizer.block_execution
 
-        # Initialize visualizer
-        self.vis = self.o3d.visualization.VisualizerWithKeyCallback()
-        self._register_key_callbacks()
-        self._initialize_visualizer()
 
-        # Visualization options
-        self.render_map = True
-        self.render_source = True
-        self.render_keypoints = False
-        self.global_view = False
-        self.render_trajectory = True
-        # Cache the state of the visualizer
-        self.state = (
-            self.render_map,
-            self.render_keypoints,
-            self.render_source,
-        )
+def screenshot_callback():
+    # TODO: this is just for demo, set a more valid path
+    if Kissualizer.polyscope.imgui.Button("SCREENSHOT"):
+        image_filename = "screenshot.jpg"
```

Comment:
How about using the current date and time in the filename? The path is fine for me. Otherwise we can also save it to the results folder.

Reply:
I was thinking about the results folder, probably better, just the path need to be computed in the pipeline also. I'll try to do it